src/ansys/fluent/core/services/datamodel_se.py

```diff
@@ -1429,9 +1449,15 @@ def __call__(self, *args, **kwds) -> Any:
         Any
             Return value.
         """
-        return self.service.execute_command(
+        file_purpose = self._get_file_purpose()
+        if file_purpose == "input" and bool(self.service.remote_file_handler):
```

Comment:
@hpohekar What's the rationale for doing getting the command's file purpose in the case of datamodel commands?

Reply:
We had extracted it on command level for both settings and datamodel. We have updated it for settings and we will update this for datamodel.

Reply:
Yes I would prefer that.